packages/staking/src/governance/api.ts

```diff
@@ -0,0 +1,102 @@
+import {GOVERNANCE_ENDPOINTS} from './config'
+import {DRepId} from './types'
+import {Fetcher, fetcher} from '@yoroi/common'
+
+export type GovernanceApi = {
+  getDRepById: (drepId: DRepId) => Promise<{txId: string; epoch: number} | null>
+  getStakingKeyState: (
+    stakeKeyHash: string,
+  ) => Promise<GetStakingKeyStateResponse>
+}
+
+export const governanceApiMaker = ({
+  networkId,
+  client = fetcher,
+}: {
+  networkId: number
+  client?: Config['client']
+}) => {
+  return new Api({networkId, client})
+}
+
+class Api implements GovernanceApi {
```

Comment:
personally I don't like using classes on JS, cuz every time you find a callback, and sometimes it can be nested deep in the code, you have to `.bind` the context.